File: SimulationCpp/projektPO/zolw.cpp
#include"zolw.h"
#include <iostream>
using namespace std;

Zolw::Zolw(int x, int y, Swiat* s) {
	sila = 2;
	this->x = x;
	this->y = y;
	inicjatywa = 1;
	zycie = 1;
	this->s = s;
}

void Zolw::wypisz()const {
	SetConsoleTextAttribute(GetStdHandle(STD_OUTPUT_HANDLE), KOLORZOLWIA);
	cout << "8";
}

void Zolw::akcja() {
	int nX = x, nY = y;
	int i = rand() % SZANSANARUCH;
	if (i == UDANA) {
		ruch(nX, nY);
		tura(nX, nY);
	}
}

void Zolw::kolizja(int nX, int nY) {
	makeLog(s->tab[nX][nY]->wypiszNazwe(), wypiszNazwe(), s->log, to_string(nX), to_string(nY), KOMz7);
	if (s->tab[nX][nY]->getSila() < 5) {
		makeLog(wypiszNazwe(), s->tab[nX][nY]->wypiszNazwe(), s->log, to_string(x), to_string(y), KOMz3);
		return;
	}
	if (s->tab[nX][nY]->getSila() > sila || s->tab[nX][nY]->getSila() == sila) {
		zycie = 0;
		makeLog(s->tab[nX][nY]->wypiszNazwe(), wypiszNazwe(), s->log, to_string(x), to_string(y), KOMz1);
		s->tab[x][y] = s->tab[nX][nY];
		s->tab[nX][nY] = nullptr;
		s->tab[x][y]->setX(x);
		s->tab[x][y]->setY(y);
	}
	else {
		if (s->tab[nX][nY]->getSila() < sila) {
			makeLog(wypiszNazwe(), s->tab[nX][nY]->wypiszNazwe(), s->log, to_string(x), to_string(y), KOMz1);
			s->tab[nX][nY]->zycie = MARTWY;
			s->tab[nX][nY] = nullptr;
		}
	}
}

string Zolw::wypiszNazwe() const {
	return "zolw";
}
Organizm* Zolw::dziecko() {
	return new Zolw(x, y, s);
}

File: SimulationJava/src/Zolw.java
import java.util.Random;
import javax.swing.ImageIcon;
import javax.swing.JLabel;

public class Zolw extends Zwierze {
    
    static private int probaruchu=4;
    static private int udana=0;

    Zolw(int xx,int yy, Swiat s){
        x=xx;
        y=yy;
        inicjatywa=1;
        sila=2;
        
        this.s=s;
        if(s instanceof Hexagony) obrazek = new JLabel(new ImageIcon("zolwhex.png"));
        else obrazek = new JLabel(new ImageIcon("zolw.png"));
    }

    @Override
    public Organizm dziecko() {
        return new Lis(x,y,s);
    }

    @Override
    public void akcja() {
        int r;
        Random rand = new Random();
        r = rand.nextInt(probaruchu);
        if(r==udana){
            noweXY nowe = new noweXY(x,y);
            ruch(nowe);
            tura(nowe);
        }
    }

    @Override
    protected void kolizja(int nX, int nY) {
        makeLog(s.getPole(nX, nY).wypiszNazwe(), "", 3, nX, nY);
        if (s.getPole(nX, nY).getSila() < 5) {
            makeLog(wypiszNazwe(), s.getPole(nX, nY).wypiszNazwe(), 4, x, y);
            return;
        }
        if (s.getPole(nX, nY).getSila() > sila || s.getPole(nX, nY).getSila() == sila) {
            makeLog(s.getPole(nX, nY).wypiszNazwe(), wypiszNazwe(), 2, x, y);
            zycie = 0;
            s.setTab(x, y,s.getPole(nX,nY));
            s.setTab(nX, nY,null);
            s.getPole(x, y).setX(x);
            s.getPole(x, y).setY(y);
        }
        else {
            if (s.getPole(nX, nY).getSila() < sila) {
                makeLog(wypiszNazwe(), s.getPole(x, y).wypiszNazwe(), 2, x, y);
                s.getPole(nX, nY).zycie = martwy;
                s.setTab(nX, nY,null);
            }
        }
    }


    @Override
    public String wypiszNazwe() {
        return "zolw";
    }

    @Override
    public JLabel wypisz() {
        return obrazek;
    }
    
}

